modules/issue_tracker/php/uploadhelper.class.inc

```diff
@@ -0,0 +1,254 @@
+                'tmp_name'    => $files['file']['tmp_name'],
+                'size'        => $this->_bytes,
+                'inserted_by' => $this->_user->getData('UserID'),
+                'description' => $this->_values['fileDescription'] ?? '',
+            );
+
+        $this->setFullPath($this->_fileToUpload);
+
+        $this->begin();
+        $this->registerFile($this->_fileToUpload);
+        $this->endWithSuccess();
+
+        return array('success' => true);
+    }
+
+    /**
+     * Sets $fileToUpload->full_path to be the full path
+     * of where the attachment file will be stored.
+     *
+     * @param object $fileToUpload The file to upload
+     *
+     * @return void
+     * @throws \LorisException
+     */
+    function setFullPath(Object &$fileToUpload) : void
+    {
+        $config = \NDB_Config::singleton();
+        $attachment_data_dir = trim(
+            rtrim(
+                $config->getSetting('IssueTrackerDataPath'),
+                '/'
+            )
+        );
+
+        if (is_writable($attachment_data_dir)) {
+            $fileToUpload->full_path = $attachment_data_dir .
+                                       '/attachments';
+            if (!is_dir($fileToUpload->full_path)) {
+                mkdir($fileToUpload->full_path, 0770, true);
+            }
+        } else {
+            throw new \LorisException(
+                'Issue_Tracker attachments directory not writable.'
+            );
+        }
+
+    }
+
+    /**
+     * This insert a record in the mysql
+     * 'issues_attachments' table.
+     *
+     * @param object $fileToUpload The object containing
+     *                             the $_FILES and the $_POST values.
+     *
+     * @return void
+     */
+    function registerFile(Object &$fileToUpload) : void
+    {
+        $DB     =& \Database::singleton();
+        $values = array(
+            'issueID'     => $this->_values['issueID'],
+            'file_hash'   => $fileToUpload->file_hash,
+            'date_added'  => date('Y-m-d h:i:s', time()),
+            'file_name'   => $fileToUpload->file,
+            'deleted'     => 0,
+            'user'        => $fileToUpload->inserted_by,
+            'description' => $fileToUpload->description,
+            'file_size'   => $fileToUpload->size,
+            'mime_type'   => $fileToUpload->mime_type
+        );
+        try {
+            $DB->insert('issues_attachments', $values);
+            // Move file from tmp to issue_tracker attachment directory.
+            $path = $fileToUpload->full_path . '/' . $fileToUpload->file_hash;
+            move_uploaded_file($_FILES['file']['tmp_name'], $path);
+        } catch (\DatabaseException $e) {
+            $this->endWithFailure();
+            error_log("DatabaseException: $e");
+            $this->_response = json_encode(
+                array(
+                    'message' => 'File registration failed.
```

Comment:
The description can be NULL in the database. This doesn't correspond to a DB exception

Reply:
@johnsaigle I decided to remove it with new commit.